osu.Game.Rulesets.Catch/Objects/CatchHitObject.cs

```diff
@@ -159,27 +159,29 @@ public void UpdateComboInformation(IHasComboInformation? lastObj)
         {
             // Note that this implementation is shared with the osu! ruleset's implementation.
             // If a change is made here, OsuHitObject.cs should also be updated.
-            ComboIndex = lastObj?.ComboIndex ?? 0;
-            ComboIndexWithOffsets = lastObj?.ComboIndexWithOffsets ?? 0;
-            IndexInCurrentCombo = (lastObj?.IndexInCurrentCombo + 1) ?? 0;
+            int index = lastObj?.ComboIndex ?? 0;
+            int indexWithOffsets = lastObj?.ComboIndexWithOffsets ?? 0;
+            int inCurrentCombo = (lastObj?.IndexInCurrentCombo + 1) ?? 0;
 
-            if (this is BananaShower)
+            // For the purpose of combo colours, spinners never start a new combo even if they are flagged as doing so.
+            if (this is not BananaShower)
             {
-                // For the purpose of combo colours, spinners never start a new combo even if they are flagged as doing so.
-                return;
+                // At decode time, the first hitobject in the beatmap and the first hitobject after a banana shower are both enforced to be a new combo,
+                // but this isn't directly enforced by the editor so the extra checks against the last hitobject are duplicated here.
+                if (NewCombo || lastObj == null || lastObj is BananaShower)
```

Comment:
@peppy mentioned in passing on stream that he'd consider coalescing these two conditionals into a single one because the nesting makes it hard to read otherwise. I'm neither here or there on that, I could live with this version, but at least maybe it's a good thing to at least propose it.